octobot/src/server/admin.rs

```diff
@@ -37,113 +38,99 @@ pub struct RepoAdmin {
 
 
 impl UserAdmin {
-    pub fn new(config: Arc<Config>, op: Op) -> Box<UserAdmin> {
-        Box::new(UserAdmin {
+    pub fn new(config: Arc<Config>, op: Op) -> Arc<UserAdmin> {
+        Arc::new(UserAdmin {
             config: config,
             op: op,
         })
     }
 }
 
 impl RepoAdmin {
-    pub fn new(config: Arc<Config>, op: Op) -> Box<RepoAdmin> {
-        Box::new(RepoAdmin {
+    pub fn new(config: Arc<Config>, op: Op) -> Arc<RepoAdmin> {
+        Arc::new(RepoAdmin {
             config: config,
             op: op,
         })
     }
 }
 
-
+#[async_trait::async_trait]
 impl Handler for UserAdmin {
-    fn handle(&self, req: Request<Body>) -> FutureResponse {
+    async fn handle(&self, req: Request<Body>) -> Result<Response<Body>> {
         match &self.op {
-            &Op::List => self.get_all(req),
-            &Op::Create => self.create(req),
-            &Op::Update => self.update(req),
-            &Op::Delete => self.delete(req),
+            &Op::List => self.get_all(req).await,
+            &Op::Create => self.create(req).await,
+            &Op::Update => self.update(req).await,
+            &Op::Delete => self.delete(req).await,
         }
     }
 }
 
 impl UserAdmin {
-    fn get_all(&self, _: Request<Body>) -> FutureResponse {
+    async fn get_all(&self, _: Request<Body>) -> Result<Response<Body>> {
         #[derive(Serialize)]
         struct UsersResp {
             users: Vec<UserInfo>,
         }
 
-        let users = match self.config.users().get_all() {
-            Ok(u) => u,
-            Err(e) => {
-                return self.respond_error(&format!("{}", e));
```

Comment:
here was being a bit inconsistent. used to leak the error db message. 🤷 

no longer!